archinstall/lib/user_interaction.py

```diff
@@ -98,22 +98,45 @@ def sig_handler(sig :int, frame :Any) -> None:
 
 	return True
 
+def is_strong_password(passwd :str) -> bool:
+
+	if not (
+		any(char.isdigit() for char in passwd)
+		and any(char.isupper() for char in passwd)
+		and any(char.islower() for char in passwd)
+		and len(passwd) >= 8
+	):
+
+		prompt = _("Your password is too weak...\n")
+		prompt += _("Are you sure to continue? ")
+
+		choice = Menu(prompt, ['yes', 'no'], default_option='no').run()
+		if choice == 'no':
+			return False
+		return True
+
+	return True
+
 
 def get_password(prompt :str = '') -> Optional[str]:
 	if not prompt:
 		prompt = _("Enter a password: ")
 
 	while passwd := getpass.getpass(prompt):
+
+		if len(passwd.strip()) <= 0:
+			break
+
+		if not is_strong_password(passwd):
+			continue
+
 		passwd_verification = getpass.getpass(prompt=_('And one more time for verification: '))
 		if passwd != passwd_verification:
 			log(' * Passwords did not match * ', fg='red')
 			continue
 
-		if len(passwd.strip()) <= 0:
-			break
-
 		return passwd
-	return None
+	return
```

Comment:
linters and guidelines tells us to explicitly return `None` even tho they in the end does exaxtly the same thing IIRC